third-party/realsense-file/rosbag2/CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 3.10)
+project(rosbag2_independent)
+
+# Allow consumers to opt-in explicitly; default OFF to avoid configuration failures
+option(BUILD_ROSBAG2 "Build rosbag2 third-party components" OFF)
+if(NOT BUILD_ROSBAG2)
+    message(STATUS "rosbag2: BUILD_ROSBAG2=OFF - skipping build of rosbag2 third-party components")
+    return()
+endif()
+
+# Prevent duplicate processing if this directory is added more than once
+if(TARGET rcutils OR TARGET rosbag2_independent)
+    message(STATUS "rosbag2: targets already defined - skipping duplicate inclusion")
+    return()
+endif()
+
+set(CMAKE_CXX_STANDARD 14)
+
+# ---- FetchContent for direct library dependencies ----
+include(FetchContent)
+
+# yaml-cpp
+FetchContent_Declare(
+    yaml-cpp
+    GIT_REPOSITORY https://github.com/jbeder/yaml-cpp.git
+    GIT_TAG master  # using master for latest version, we can also pick a specific commit but we need one after version 0.8.0 - we need YAML_CPP_DISABLE_UNINSTALL flag
```

Comment:
We always choose a specific commit / release so our code will not break.
Please also explane is a comment the need for this package.